linkerd/app/integration/src/tests/profiles.rs

```diff
@@ -1,123 +1,142 @@
+                    .request_path("/load-profile")
+                    .label("load_profile", "test"),
+            ],
+            budget: Some(controller::retry_budget(Duration::from_secs(1), 0.1, 1)),
+            default_routes: true,
         }
-    };
-    (routes: [$($route:expr),+], budget: $budget:expr, with_client: $with_client:expr, with_metrics: $with_metrics:expr) => {
-        profile_test! {
-            http: http1,
-            routes: [$($route),+],
-            budget: $budget,
-            with_client: $with_client,
-            with_metrics: $with_metrics
+    }
+
+    fn with_budget(self, budget: impl Into<Option<controller::pb::RetryBudget>>) -> Self {
+        Self {
+            budget: budget.into(),
+            ..self
         }
-    };
-    (http: $http:ident, routes: [$($route:expr),+], budget: $budget:expr, with_client: $with_client:expr, with_metrics: $with_metrics:expr) => {
-        let _trace = trace_init();
+    }
 
-        let counter = AtomicUsize::new(0);
-        let counter2 = AtomicUsize::new(0);
-        let counter3 = AtomicUsize::new(0);
-        let host = "profiles.test.svc.cluster.local";
+    fn with_profile_route(self, route: controller::RouteBuilder) -> Self {
+        let mut routes = self.routes;
+        routes.push(route);
+        Self { routes, ..self }
+    }
 
-        let srv = server::$http()
+    async fn run(self) -> Test {
+        let (trace, _) = trace_init();
+        let host = "profiles.test.svc.cluster.local";
+        let mut srv = self
+            .server
             // This route is just called by the test setup, to trigger the proxy
             // to start fetching the ServiceProfile.
             .route_fn("/load-profile", |_| {
-                Response::builder()
-                    .status(201)
-                    .body("".into())
-                    .unwrap()
-            })
-            .route_fn("/1.0/sleep",  move |_req| {
-                ::std::thread::sleep(Duration::from_secs(1));
-                Response::builder()
-                    .status(200)
-                    .body("slept".into())
-                    .unwrap()
-            })
-            .route_async("/0.5",  move |req| {
-                let fail = counter.fetch_add(1, Ordering::Relaxed) % 2 == 0;
-                async move {
-                    // Read the entire body before responding, so that the
-                    // client doesn't fail when writing it out.
-                    let _body = hyper::body::to_bytes(req.into_body()).await;
-                    tracing::debug!(body = ?_body.as_ref().map(|body| body.len()), "recieved body");
-                    Ok::<_, Error>(if fail {
+                Response::builder().status(201).body("".into()).unwrap()
+            });
+        if self.default_routes {
+            let counter = AtomicUsize::new(0);
+            let counter2 = AtomicUsize::new(0);
+            let counter3 = AtomicUsize::new(0);
```

Comment:
I know this isn't new code but it's a bit weird that these endpoints alternate between failure and success without any explicit control